cpp/tests/datetime/datetime_ops_test.cpp

```diff
@@ -52,16 +52,6 @@ TYPED_TEST(NonTimestampTest, TestThrowsOnNonTimestamp)
   cudf::data_type dtype{cudf::type_to_id<T>()};
   cudf::column col{dtype, 0, rmm::device_buffer{}, rmm::device_buffer{}, 0};
 
-  EXPECT_THROW(extract_year(col), cudf::logic_error);
```

Comment:
I don't think we should delete tests until the corresponding code is removed.

Reply:
We should delete the tests but in this case they should be replaced with the equivalent calls to the new API in this PR.

Reply:
Done.

Reply:
This is marked as resolved but I don't see any equivalent test in this file. We want a test that ensures the new date extraction function throws a `cudf::logic_error` if given a non-timestamp input. If I'm just not seeing this, can you link to it in the source?

Reply:
Ah you're right.

Reply:
Added those tests back.